lambdas/config.py

```diff
@@ -83,14 +83,15 @@ def validate_input(input_data: dict) -> None:
     # If next step is extract step, required harvest fields are present
     # ruff: noqa: SIM102
     if input_data["next-step"] == "extract":
-        if missing_harvest_fields := [
-            field for field in REQUIRED_HARVEST_FIELDS if field not in input_data
-        ]:
-            message = (
-                "Input must include all required harvest fields when starting with "
-                f"harvest step. Missing fields: {missing_harvest_fields}"
-            )
-            raise ValueError(message)
+        if input_data["source"] not in ["gismit", "gisogm"]:
+            if missing_harvest_fields := [
+                field for field in REQUIRED_OAI_HARVEST_FIELDS if field not in input_data
+            ]:
+                message = (
+                    "Input must include all required harvest fields when starting with "
+                    f"harvest step. Missing fields: {missing_harvest_fields}"
+                )
+                raise ValueError(message)
```

Comment:
While this definitely works, just noting a lot of nested `if` statements here.  However, because the OAI Harvester is the "default", you kind of do need a negation check here.

Ultimately, I could see validation functions for `next-step`, `run-type`, and then for `extract` it could hand off to validators depending on the harvester.  

However, not sure that's worth the effort now.  If we are going to explore unifying the TIMDEX sources across parts of the pipeline, where this lambda is a critical piece of that, might be better to leave as-is -- which is readable and easy to scan -- and consider refactoring down the road.

Reply:
**Note:** Will leave this comment 'Unresolved' for easier location